sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/CustomDeclarationOrderCheck.java

```diff
@@ -596,37 +596,27 @@ private int getPositionInOrderDeclaration(final DetailAST ast)
             if (currentRule.getClassMember() == ast.getType()
                     && currentRule.getRegexp().matcher(modifiers).find())
             {
-                if (currentRule.hasRule(ANNON_CLASS_FIELD_MACRO)) {
-                    if (isAnonymousClassField(ast)) {
-                        result = index;
-                        break;
-                    }
-                }
-                else if (currentRule.hasRule(GETTER_SETTER_MACRO)) {
+                if (currentRule.hasRule(ANNON_CLASS_FIELD_MACRO)
+                        || currentRule.hasRule(GETTER_SETTER_MACRO)
+                        || currentRule.hasRule(MAIN_METHOD_MACRO)) {
+
                     final String methodName = getIdentifier(ast);
                     final ClassDetail classDetail = classDetails.peek();
-                    if (classDetail.containsGetter(methodName)
-                            || classDetail.containsSetter(methodName))
-                    {
-                        result = index;
-                        break;
-                    }
-                }
-                else if (currentRule.hasRule(MAIN_METHOD_MACRO)) {
-                    if (isMainMethod(ast)) {
+
+                    if (isAnonymousClassField(ast)
+                            || classDetail.containsGetter(methodName)
+                            || classDetail.containsSetter(methodName)
+                            || isMainMethod(ast)) {
                         result = index;
                         break;
```

Comment:
After reducing the number of breaks to 2, not sure how to eleminate the last one ...
Maybe completly refactoring this method ...